renterhost/session.go

```diff
@@ -193,9 +196,17 @@ func (s *Session) readMessage(obj ProtocolObject, maxLen uint64) error {
 // WriteRequest sends an encrypted RPC request, comprising an RPC ID and a
 // request object.
 func (s *Session) WriteRequest(rpcID Specifier, req ProtocolObject) (err error) {
-	err = errors.Wrap(s.writeMessage(&rpcID), "WriteRequestID")
-	if err == nil && req != nil {
-		err = errors.Wrap(s.writeMessage(req), "WriteRequest")
+	err = s.writeMessage(&rpcID)
+	if err != nil {
+		err = fmt.Errorf("WriteRequestID: %w", err)
+		return
+	}
+	if req == nil {
+		return
+	}
+	err = s.writeMessage(req)
+	if err != nil {
+		err = fmt.Errorf("WriteRequest: %w", err)
```

Comment:
This is the kind of thing to look for.

`errors.Wrap()` made it easy to execute a function (`s.writeMessage()`) while ensuring `err` will remain `nil` (if no error is encountered by `s.writeMessage()`).

`errors.Wrap()` can't be easily replaced with a `fmt.Errorf()` in cases like this.

Reply:
yep. I definitely got a little "terse-happy" here. I would rewrite this without named return:

```go
if err := s.writeMessage(&rpcID); err != nil {
    return fmt.Errorf("WriteRequestID: %w", err)
}
if req != nil {
    if err := s.writeMessage(req); err != nil {
        return fmt.Errorf("WriteRequest: %w", err)
    }
}
return nil
```